src/ivc/cyclefold/incrementally_verifiable_computation/mod.rs

```diff
@@ -52,10 +74,522 @@ where
     CSup::Scalar: PrimeFieldBits + FromUniformBytes<64>,
 {
     pub fn new(
-        _pp: &PublicParams<ARITY, ARITY, CMain, CSup, SC>,
-        _sc: &SC,
-        _z_0: [CMain::ScalarExt; ARITY],
-    ) -> Self {
-        todo!("temporarily removed for the purposes of a simple merge into main")
+        pp: &mut PublicParams<ARITY, CMain, CSup, SC>,
+        sc: &SC,
+        z_0: [CMain::ScalarExt; ARITY],
+    ) -> Result<Self, Error<CMain>> {
+        let _span = info_span!("ivc_new", step = 0).entered();
+
+        let primary_initial_acc = ProtoGalaxy::<CMain, 1>::new_accumulator(
+            AccumulatorArgs::from(&pp.primary_S),
+            &pp.protogalaxy_prover_params(),
+            &mut ro(),
+            pp.primary_initial_trace.clone(),
+        )
+        .map_err(Error::WhileProtoGalaxyAccCreation)?;
+
+        // At zero step cyclefold ivc - output protogalaxy-accumulator is input
+        // protogalaxy-accumulator. Bug proof still should be valid.
+        let mut random_oracle = ro();
+        let (_new_acc, self_proof) = ProtoGalaxy::prove(
+            &pp.primary_ck,
+            &pp.protogalaxy_prover_params(),
+            &mut random_oracle,
+            primary_initial_acc.clone(),
+            &[pp.primary_initial_trace.clone()],
+        )?;
+
+        #[cfg(test)]
+        {
+            ProtoGalaxy::<CMain, 1>::is_sat(&pp.primary_ck, &pp.primary_S, &_new_acc)
+                .expect("initial primary accumulator not corrent");
+
+            assert_eq!(
+                ProtoGalaxy::verify(
+                    &pp.protogalaxy_verifier_params(),
+                    &mut ro(),
+                    &mut ro(),
+                    &primary_initial_acc.clone().into(),
+                    &[pp.primary_initial_trace.u.clone()],
+                    &self_proof,
+                )
+                .expect("while verification after first prove"),
+                _new_acc.into()
+            );
+        }
+
+        let support_initial_acc = nifs::sangria::accumulator::RelaxedPlonkTrace::from_regular(
+            pp.support_initial_trace.clone(),
+            SupportCircuit::<CMain>::MIN_K_TABLE_SIZE as usize,
+        );
+
+        // At zero step cyclefold ivc - output sangria-accumulator is input
+        // sangria-accumulator. Bug proofs still should be valid.
+        let SupportCircuitFoldResult {
+            new_accumulator: _new_new_accumulator,
+            incoming: support_incoming,
+        } = fold_support_circuit::<CMain, CSup>(
+            &pp.support_ck,
+            &pp.sangria_prover_params(),
+            &support_initial_acc,
+            primary_initial_acc
+                .u
+                .W_commitments
+                .iter()
+                .copied()
+                .zip_eq(pp.primary_initial_trace.u.W_commitments.iter().copied()),
```

Comment:
what are the definition of `primary_initial-acc` and `pp.primary_initial_trace`, 
are these two the same?

Reply:
Initial acc is accumulator, it's created from initial trace,